internal/infrastructure/kubernetes/proxy/resource_provider.go

```diff
@@ -136,6 +144,10 @@ func (r *ResourceRender) OwnerReferences() []metav1.OwnerReference {
 
 // ServiceAccount returns the expected proxy serviceAccount.
 func (r *ResourceRender) ServiceAccount() (*corev1.ServiceAccount, error) {
+	if r.usingCustomServiceAccountName() {
```

Comment:
can you help explain the need for this

Reply:
Oh, this's from the original idea, don't create sa when using custom service account name.

Reply:
sure, will do.